core/src/main/java/com/linecorp/armeria/server/logging/LoggingService.java

```diff
@@ -56,17 +56,23 @@ public static LoggingServiceBuilder builder() {
 
     LoggingService(HttpService delegate, LogWriter logWriter,
                    Sampler<? super ServiceRequestContext> successSampler,
-                   Sampler<? super ServiceRequestContext> failureSampler) {
+                   Sampler<? super ServiceRequestContext> failureSampler,
+                   Sampler<Long> slowRequestSampler) {
         super(requireNonNull(delegate, "delegate"));
         this.logWriter = requireNonNull(logWriter, "logWriter");
         requireNonNull(successSampler, "successSampler");
         requireNonNull(failureSampler, "failureSampler");
         sampler = requestLog -> {
             final ServiceRequestContext ctx = (ServiceRequestContext) requestLog.context();
+            final boolean isSlow = slowRequestSampler.isSampled(requestLog.totalDurationNanos());
+            final boolean successOrFailure;
             if (ctx.config().successFunction().isSuccess(ctx, requestLog)) {
-                return successSampler.isSampled(ctx);
+                successOrFailure = successSampler.isSampled(ctx);
+            } else {
+                successOrFailure = failureSampler.isSampled(ctx);
             }
-            return failureSampler.isSampled(ctx);
```

Comment:
If `isSlow` is true, can we skip the additional samplings?

Reply:
Samplers are stateful, if we short cut, it would mean sampler won't record the value. I.e. counting sampler won't count actual values.

Reply:
There are tests that verify this there is similar behavior here? 

https://github.com/line/armeria/blob/01847b87f703a876930f0daba6b4683c8d05c511/core/src/test/java/com/linecorp/armeria/common/util/SamplerTest.java#L139-L158